usecases/scheduled_execution/batch_filtering.go

```diff
@@ -0,0 +1,185 @@
+package scheduled_execution
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/checkmarble/marble-backend/models"
+	"github.com/checkmarble/marble-backend/models/ast"
+	"github.com/checkmarble/marble-backend/pure_utils/duration"
+
+	"github.com/jackc/pgx/v5"
+)
+
+func selectFiltersFromTriggerAstRootAnd(node ast.Node, table models.TableIdentifier) []models.Filter {
+	if node.Function != ast.FUNC_AND {
+		return nil
+	}
+
+	filters := make([]models.Filter, 0, 10)
+	for _, node := range node.Children {
+		filter, valid := astConditionToFilter(node, table)
+		if valid {
+			filters = append(filters, filter)
+		}
+	}
+	return filters
+}
+
+func astConditionToFilter(node ast.Node, table models.TableIdentifier) (models.Filter, bool) {
+	switch node.Function {
+	case ast.FUNC_GREATER,
+		ast.FUNC_GREATER_OR_EQUAL,
+		ast.FUNC_LESS,
+		ast.FUNC_LESS_OR_EQUAL,
+		ast.FUNC_EQUAL,
+		ast.FUNC_NOT_EQUAL:
```

Comment:
Worst formatting. Not a usefull comment but I needed to tell it

As a side note, I think we should extract this subset of functions as a "function". So something like :
```go
if isMathOperator(....)

if isListOperator(...)
````

It's something we already have in the frontend and it helps to share behaviours when adding new functions.

I must admit, if the one we add is not compatible with this logic it will end bad ^^'
But if it does (= the group is usefull) it helps

Reply:
Yea formatting was not kind here. I still like it better like this than together in two lines.

I get the idea of extracting, though in this case it would break the usage of switch (or I must implement a function that maps functions to a type (implying only one)... maybe there is an elegant way around this but I think I'll pass it for now.

Reply:
(I also think it's unlikely that we add more math comparison operators)

Reply:
ok I did something like this in models/scheduled_scenario_executions.go